### ISSUE TITLE
[Problem/Bug]: My Construct 3 game does not work on Xbox.

### ISSUE THREAD

**Issue by Pedro1234-code** [MicrosoftEdge/WebView2Feedback#4370]
### What happened?

Since WebView2 released on Xbox I tried to port my game made using Construct 3 for it. It always worked fine on PC, and still does, but never worked on Xbox. It launches but the launch screen is empty. The only thing that appears is the "Play" button and the background music that plays normally. Once the Construct 3 beta r378 was released (this version officially adds support for Xbox/UWP WebView2 export) I tried to export by the official methods but I still got the same results.

Note that I also tried to launch the game in the Edge browser on Xbox through itch.io and the same thing happened.

### Importance

Blocking. My app's basic functions are not working due to this issue.

### Runtime Channel

Stable release (WebView2 Runtime)

### Framework

WinUI2/UWP

### Operating System

Xbox

### OS Version

10.0.22621

### Repro steps

1. Export the game in Construct 3
2. Build with VS2022 and deploy it into Dev Mode
3. Launch it on the console.

### Repros in Edge Browser

Yes

### Regression

No, this never worked

**Comment by victorhuangwq** [comment]
@AshleyScirra would this issue be more relevant to Construct 3?

**Comment by AshleyScirra** [comment]
This was also reported to us directly here: https://github.com/Scirra/Construct-bugs/issues/7823

Here is a Visual Studio solution that can be tested: [Viajante Espacial.zip](https://github.com/MicrosoftEdge/WebView2Feedback/files/14268447/Viajante.Espacial.zip)

I cannot reproduce this on my Xbox One dev kit: the game displays identically for me both when run on Windows, and when using the "Remote machine" option to launch directly on the dev kit. However I was testing on an Xbox One, and on the issue filed with us they noted they were using Xbox Series S.

The described result would appear to suggest that the WebGL-rendered canvas element remains black, while other HTML elements (e.g. the button) and audio playback continue to work normally. This could potentially be a bug in the WebGL implementation in WebView2 on the Xbox Series S specifically. The affected user also notes that it works OK on Windows, and only exhibits a problem on Xbox Series S, which is usually good evidence of a problem with WebView2 on Xbox rather than any problem with Construct itself.

I don't currently have an Xbox Series S, and would rather not order one in given the likelihood that this is a problem with WebView2 and not Construct, so I would recommend Microsoft try reproducing this by running the Visual Studio solution I provided on an Xbox Series S.

**Comment by Pedro1234-code** [comment]
Note that the same thing happens in Edge browser on Xbox.

**Comment by victorhuangwq** [comment]
Thanks for looking into it @AshleyScirra. @jennifer-jacobs could you help see if the issue reproduces, specifically on the Series S? Seems like it does not repro on Xbox One. Do let me know if someone else is better suited for looking into it.

**Comment by Pedro1234-code** [comment]
Just to add that the full game is now released on Microsoft Store for Xbox. I decided to release so more people could test without Dev Mode. The name is "Viajante do Espaço".

https://www.microsoft.com/store/productId/9MTCN09D44TL

**Comment by Pedro1234-code** [comment]
Heard from another person with Series S that same thing is happening to her. I also want to add that I'm not able to play preview on Construct 3 on Xbox Edge. The entire screen goes black for a second and then the tab is closed.

**Comment by jennifer-jacobs** [comment]
Hi @Pedro1234-code,
Which API specifically doesn't work on Series S? Can you point me to the code in your sample app that has the difference in behavior?

Also, I'm curious if this repros on a Series X (if you have one)?

**Comment by Pedro1234-code** [comment]
> Hi @Pedro1234-code,
> Which API specifically doesn't work on Series S? Can you point me to the code in your sample app that has the difference in behavior?
> 
> Also, I'm curious if this repros on a Series X (if you have one)?

No I don't have an Xbox Series X, only the S sadly. I can't exactly point which API does not work, but I released the game on Microsoft Store to make it easier for testing (and heard from another Series S user that the same thing happened to her). If you want to test the game you can just download it, it's called "Viajante do Espaço" in portuguese. I can also send you the full project file but I don't think it's needed as there is already a stripped down version of it here. I also wanted you to understand that I don't have so much experience with coding and me and my team decided to use Construct specifically because it's more graphical and beginner friendly. 

Also wanted to add that I'm 99% sure that it has some relation with the other bug where you can't play Construct 3 games on Edge. I think it's not related to WebView2 but more with the rendering engine. Something is telling me that this could potentially be some bug in the GPU driver of the Series S, specially if it does not reproduce on the Series X.

**Comment by Pedro1234-code** [comment]
@jennifer-jacobs I can confirm that the issue reproduces on Series X. Got the confirmation from someone at Reddit that the same thing occurs on Series X.

**Comment by jennifer-jacobs** [comment]
I was also able to confirm this problem repros on a Series X.  It's been added to our backlog!

**Comment by thejacs** [comment]
Hi guys. Same problem testing on **Xbox Series X**

It works well on Windows, but when testing on Xbox Series X it just start in a black screen and later change to a white screen and gets frozen there forever (could be the background of my first scene, but not sure...).

My doubt is if this is only when testing through Visual Studio (Remote machine) or if the compilation would fail also with a native package executed from Xbox... 

If this is an Xbox problem... Does anybody knows if Microsoft is aware of it?

**Comment by thejacs** [comment]
Hi @jennifer-jacobs, How is this problem going on? Do you think guys it will be solved soon?

Also I guess, for what @Pedro1234-code said, it wouldn't make sense to upload a game to Xbox yet because the problem goes beyond the remote machine, is this right?

**Comment by Pedro1234-code** [comment]
@thejacs Didn't exactly understand your question. I already uploaded to Microsoft Store/Xbox Retail Mode as it's working perfectly on Xbox One but on Xbox Series it just shows the black screen.

**Comment by thejacs** [comment]
@Pedro1234-code, It was actually a question for Jennifer, sorry. I just meant that if the problem is not the remote machine (as you confirmed) it doesn't make sense to upload more games on Xbox Store for now, until they fix the problem on Series X/Y.

**Comment by Pedro1234-code** [comment]
I uploaded first because it's working on Xbox One and second because it's easier for people to test.

**Comment by thejacs** [comment]
I know @Pedro1234-code What you did has been very useful. I said "more" games.

**Comment by thejacs** [comment]
Hi @victorhuangwq, is there any advance with this problem? 
Do you know when (more or less) will be able to upload UWP (WebView2) games to run on Xbox Series?

Thank you.

**Comment by victorhuangwq** [comment]
Hi @thejacs do you know what API is being called that is producing this unexpected result?

**Comment by victorhuangwq** [comment]
@thejacs and @Pedro1234-code I'm wondering if https://webglsamples.org/aquarium/aquarium.html works for you on Edge on Xbox Series?

**Comment by thejacs** [comment]
Thanks for your concern @victorhuangwq , I'll test it today later on my Xbox Series X and let you know asap.

**Comment by thejacs** [comment]
Hi @victorhuangwq . Yes, the aquarium sample works fine on Edge in my Xbox Series X.

**Comment by thejacs** [comment]
On the other hand, @victorhuangwq @AshleyScirra , I tried to open "Ghost Racer" (an example Construct 3 game) and it doesn't work on Xbox Series X Edge either (of course it works on PC Edge).

**Comment by victorhuangwq** [comment]
@thejacs could you get a dump of the browser and GPU process when you see the black screen if you know how to?

@AshleyScirra if the WebGL Aquarium works on the Xbox Series X, I don't want to entirely rule out the possibility of it possibly being a Construct 3 issue, or at least how it uses WebGL.

**Comment by thejacs** [comment]
@victorhuangwq As Pedro, I just develope indie games with Construct 3 and I'm not an expert on this area. But I can try if you explain to me what to do.  I'll be glad to help.

**Comment by AshleyScirra** [comment]
I can't completely rule out there being some problem with Construct, but given that it works on all other platforms, usually this type of problem works out as a graphics driver bug, or a bug in the browser WebGL implementation.

**Comment by victorhuangwq** [comment]
We are still investigating this issue; we are indeed suspecting something to do with Xbox graphics driver bug.

**Comment by thejacs** [comment]
Thank you @victorhuangwq, I'm sure you guys will find the bug. 
Please let us know when we'll be able tu run Construct 3 Games on Xbox Series :D

**Comment by victorhuangwq** [comment]
No problem. Thanks for the patience, we provide any updates here.

**Comment by victorhuangwq** [comment]
We have some parties looking into this issue who are currently out of the office, we will circle back to this when they are back.

**Comment by AshleyScirra** [comment]
I ordered in an Xbox Series S and I'm now able to reproduce this. The canvas content is blank when run on the Xbox Series S specifically, but works correctly on Windows.

I managed to attach Edge DevTools to the Xbox Series S and it logs some errors - firstly from our engine with a "framebuffer incomplete" error, and secondly a "context lost" event.

Overall I'd say this looks a lot like a graphics driver bug - I'll tinker with it and see if we can come up with a workaround, but dealing with graphics driver bugs can be really hard going.

**Comment by AshleyScirra** [comment]
I think I figured it out! The problem is on Xbox Series S, the WebGL extensions `EXT_disjoint_timer_query_webgl2` and `EXT_disjoint_timer_query` are broken. Using these extensions appears to cause the WebGL context to crash. In Construct I've worked around the problem by making sure these extensions are not used in Xbox exports. The workaround is in the next Construct release (r394+) which should unblock publishing to Xbox for Construct users.

Of course, it would be great if Microsoft could still fix this - either return `null` for unsupported extensions, or better still, make sure those extensions work.

**Comment by victorhuangwq** [comment]
Thank you @AshleyScirra for figuring the issue out! I am following up with my team on this:
> Of course, it would be great if Microsoft could still fix this - either return null for unsupported extensions, or better still, make sure those extensions work.

Does running https://registry.khronos.org/webgl/sdk/tests/conformance/extensions/ext-disjoint-timer-query.html on your Xbox Series S crash as well?

**Comment by thejacs** [comment]
Very good news! Thank you @AshleyScirra. Once (r394+) will be available,  beta or stable, I will export my game again and try it on Xbox Series X too, so I can confirm you guys if it also works. Definately, it should :-)

**Comment by Pedro1234-code** [comment]
@AshleyScirra That's amazing news! Thanks for your dedication on solving this problem!

**Comment by AshleyScirra** [comment]
Just an update, the workaround is in Construct 3 [r394](https://www.construct.net/en/make-games/releases/beta/r394) (which came out last week) and newer, so it should be possible to publish to Xbox using those releases.

**Comment by thejacs** [comment]
Hello @AshleyScirra, I finally tested my game on Xbox Series X exported with r395.
I can confirm you that it now runs on Xbox Series X, butI had some issues. Let me share:

My game has 3 characters that you can switch and the sprites just appear and disappear randomly, specially when I control them and they are in movement (they work with platform behaviour).

Less important but want to mention: Pad control is not clean on Xbox and weird things happens (on Windows and Linux the same Xbox pad works perfect):

The sticks axis look to have different sensibility. For example, I have to move left stick very little or the Scroll behaviour gets lost (not platform behaviour, what is weird). I mean, If I use left stick full left, the characters walk to the left perfectly but the "camera" doesn't follow him. If I just push it a little bit to left it works well.

Same happen with menus, I have to move it just a little up or down to move in a menu properly.


I hope this helps.

(Note: I don't know if those errors could be caused by Visual Studio/Virtual machine mode, but shouldn't).

**Comment by AshleyScirra** [comment]
@thejacs - this issue is about a specific problem with Microsoft WebView2. If you believe there are different problems with Construct, please file an issue on our own tracker following all the guidelines, as we need all that information to be able to help: https://github.com/Scirra/Construct-bugs

**Comment by thejacs** [comment]
Hi @AshleyScirra, I'm sure you are right (you are the expert on this area). I just wanted to give you guys feedback (also @victorhuangwq) about what happened after to export my game with r395 and run it on Xbox Series X.

Perhaps it helps...
Regards.

**Comment by thejacs** [comment]
Hi guys @victorhuangwq @AshleyScirra, I have been doing several tests more on Xbox Series X.
Yes, I can confirm you that it now runs on the console, but **definately the bug is still there** somewhere. 
There are **objects that randomly appear or dissapear**. And not only sprites, also, for example, Text objetcs.
I hope this info helps to solve the problem and to have Construct 3 Games 100% playable on Xbox Series soon.

**Comment by thejacs** [comment]
Hi @victorhuangwq, did you guys find the bug? It looks that Construct 3 games work properly in all platforms (even on desktop Edge or Xbox One) but not in Xbox Series X/Y. Let us know, please. It's taking too long :-(

**Comment by victorhuangwq** [comment]
So, I am assuming this issue does not replicate in the Edge Browser? 

If so, could you try https://registry.khronos.org/webgl/sdk/tests/conformance/extensions/ext-disjoint-timer-query.html and see if this works on the Edge browser in Xbox Series X?

**Comment by thejacs** [comment]
Hi @victorhuangwq I made the test you asked for, and definately there are 3 errors only on Xbox Series X/S Edge.
I also tried the test on 2 Desktop computer's Edge and even on Android with no errors. 
So 100% is a Xbox Series X/S Edge bug. Let me know if you need anything else.


![Test](https://github.com/user-attachments/assets/b5e01d82-de9d-492c-b29c-42c16d6712e0)

![Errors](https://github.com/user-attachments/assets/4ad565ef-162a-4265-88ce-deb6da7b5c86)

**Comment by thejacs** [comment]
Hello @victorhuangwq, did you have time to check the 3 EDGE Fails that I posted? @AshleyScirra let us know what you think, please. I know is an EDGE bug, but maybe your opinion helps here. Thank you, guys.

**Comment by AshleyScirra** [comment]
There is a bug in the Xbox system that the latest Construct release has already got a workaround for. If you are using the latest release and there is still a problem, I would strongly advise to file a separate issue, as it is unlikely to be related to this issue, and trying to discuss it here will be confusing and make it harder to solve both problems.

**Comment by thejacs** [comment]
If you recomend that, I will start a new Issue, no problem. Yes, I even tried yesterday R403 Construct version, and still there are images and texts that just vanish (on Xbox Series X/Y). So definately is an Xbox bug.
